bentoml/keras.py

```diff
@@ -187,39 +176,59 @@ def __init__(
             tag=tag,
             predict_fn_name=predict_fn_name,
             device_id=device_id,
+            partial_kwargs=partial_kwargs,
             resource_quota=resource_quota,
             batch_options=batch_options,
             model_store=model_store,
         )
         self._session = _sess
 
-    # pylint: disable=arguments-differ,attribute-defined-outside-init
-    def _setup(self) -> None:  # type: ignore[override] # noqa
+    # pylint: disable=attribute-defined-outside-init
+    def _setup(self) -> None:
         self._session.config = self._config_proto
         self._model = load(self.name, model_store=self._model_store)
-        self._predict_fn = getattr(self._model, self._predict_fn_name)
-
-    # pylint: disable=arguments-differ
-    def _run_batch(  # type: ignore[override]
-        self, input_data: t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor]
-    ) -> t.Union[np.ndarray, tf.Tensor]:
-        if not isinstance(input_data, (np.ndarray, tf.Tensor)):
-            input_data = np.array(input_data)
-        with self._device:
+        raw_predict_fn = getattr(self._model, self._predict_fn_name)
+        self._predict_fn = functools.partial(raw_predict_fn, **self._partial_kwargs)
+
+    def _run_batch(
+        self,
+        *args: t.Union[
+            t.List[t.Union[int, float]], "np.ndarray[t.Any, np.dtype[t.Any]]", tf.Tensor
+        ],
+        **kwargs: t.Any,
```

Comment:
@aarnphm  does keras really need this? The `predict` should only accept one input (multiple inputs are inside a list). So we may just need partial kwargs. And here partial kwargs is not flexible like in pytorch's case, just some configuration parameter for keras's predict function. So in my pr I renamed it into predict_kwargs (#2069)

Reply:
noted. Was trying to make pyright happy :)